activelearning/surrogate/feature_extractor/ocp.py

```diff
@@ -0,0 +1,68 @@
+import torch.nn as nn
+import torch_geometric
+from activelearning.surrogate.feature_extractor.mlp import MLP
+from ocpmodels.common.utils import make_trainer_from_dir
+from ocpmodels.common.gfn import FAENetWrapper
+from ocpmodels.datasets.data_transforms import get_transforms
+from torch_scatter import scatter
+
+
+class FAENetFeatureExtractor(nn.Module):
+    def __init__(
+        self, checkpoint_path, device, n_hidden=[32, 64], n_output=2, float_precision=32
+    ):
+        super(FAENetFeatureExtractor, self).__init__()
+        self.device = device
+        # have to set the ROOT to a folder that contains "configs/models/faenet.yaml" and "configs/models/tasks/is2re.yaml"
+        # abs_config_dir = os.path.abspath("config/")
+        # ocp_utils.ROOT = Path(abs_config_dir + "/surrogate/faenet").resolve()
+        # ocp_utils.ROOT = Path(get_original_cwd() + "/config/surrogate/faenet").resolve()
+
+        trainer = make_trainer_from_dir(
+            checkpoint_path,
+            mode="continue",
+            overrides={
+                "is_debug": True,
+                "silent": True,
+                "cp_data_to_tmpdir": False,
+                "deup_dataset.create": False,
+                "model": {
+                    "dropout_lin": 0.0
+                },  # for inference, we don't want to have dropout
+                "cpu": device == "cpu",
+            },
+            skip_imports=["qm7x", "gemnet", "spherenet", "painn", "comenet"],
+            silent=True,
+        )
+        trainer.load_checkpoint(checkpoint_path)
```

Comment:
This is unfortunately sth that has to stay, because the "make_trainer_from_dir" loads the last checkpoint of this directory and not the best checkpoint. Victor found this out and proposed this fix